File: code/fileserver/fileserver.go
/**
 * go build fileserver.go
 * ./fileserver -p=8100 -d="."
 */

package main

import (
	"flag"
	"net/http"
	"log"
)

func cors_wrapper(handler http.Handler) func(http.ResponseWriter, *http.Request) {
	return func(w http.ResponseWriter, r *http.Request) {
		w.Header().Set("Access-Control-Allow-Origin", "*")
		handler.ServeHTTP(w, r)
	}
}

func main() {
	port := flag.String("p", "8100", "port to serve on")
	directory := flag.String("d", ".", "the directory of static file to host")
	flag.Parse()
	http.HandleFunc("/", cors_wrapper(http.FileServer(http.Dir(*directory))))
	log.Printf("Serving %s on HTTP port: %s\n", *directory, *port)
	log.Fatal(http.ListenAndServe(":" + *port, nil))
}

File: code/fileserver/fileserver.js

const express = require('express');
const serveIndex = require('serve-index');
const path = require('path');
const minimist = require('minimist');

const app = express()

const args = minimist(process.argv.slice(2), {
    alias: {
        h: 'help',
        p: 'port',
        d: 'directory'
    },
    default: {
        port: 8100,
        directory: '.'
    }
});
if (args.help) {
    console.log(`A simple file server with CORS
      -h,--help\thelp
      -p,--port\tport (default:8100)
      -d,--directory\tbase directory (default: .)`)
    process.exit()
}

const get_path = p => {
    if (path.isAbsolute(p)) return p;
    return path.join(__dirname, p);
};

const port = args['port'];
const dir = get_path(args['directory'])

const setHeaders = (res, path, stat) => {
    res.set('Access-Control-Allow-Origin', '*');
};

app.use(express.static(dir, {setHeaders}))
app.use(serveIndex(dir))

app.listen(port, () => console.log(`File server listening on port ${port}`))
